internal/util/errors.go

```diff
@@ -66,5 +72,31 @@ func ShouldRetryVolumeGeneration(err error) bool {
 	return (errors.Is(err, ErrKeyNotFound) ||
 		errors.Is(err, ErrPoolNotFound) ||
 		errors.Is(err, ErrImageNotFound) ||
+		errors.Is(err, rados.ErrPermissionDenied) ||
+		errors.Is(err, ErrRBDGroupNotFound))
+}
+
+// ShouldRetryVolumeGroupGeneration determines whether the process of finding or generating
+// volumegroups should continue based on the type of error encountered.
+//
+// It checks if the given error matches any of the following known errors:
+//   - ErrPoolNotFound: The rbd pool where the volumegroup/omap is expected doesn't exist.
+//   - ErrRBDGroupNotFound: The volumegroup doesn't exist in the rbd pool.
+//   - rados.ErrPermissionDenied: Permissions to access the pool is denied.
```

Comment:
Why retry on PermissionDenied? This usually is a pretty final error.

Reply:
Thanks for the pointer. That commit message contains a description of the _why_, ideally that is included in the comment, but in the commit message it is ok too.